service/src/main/java/bio/terra/workspace/service/resource/model/WsmResourceHandler.java

```diff
@@ -23,10 +25,24 @@ public interface WsmResourceHandler {
   /**
    * Generate the resource cloud-native name for resource
    *
-   * @param workspaceUuid workspace UUID, when it is not null, the generated name will attach
-   *     workspace project id.
+   * @param workspaceUuid workspace UUID
    * @param resourceName resource name
    * @return cloud-native name
    */
-  String generateCloudName(@Nullable UUID workspaceUuid, String resourceName);
+  default String generateCloudName(@Nullable UUID workspaceUuid, String resourceName) {
+    throw new BadRequestException(
+        "generateCloudName with workspaceUuid and resourceName not supported");
+  }
+
+  /**
+   * Generate the resource cloud-native name for resource
+   *
+   * @param workspaceUserFacingId workspace UserFacingId
+   * @param resourceName resource name
+   * @return cloud-native name
+   */
+  default String generateCloudName(@NotNull String workspaceUserFacingId, String resourceName) {
```

Comment:
rm notnull.

Reply:
handled in https://github.com/DataBiosphere/terra-workspace-manager/pull/1227